### PR TITLE
Add metadata columns

### PR DESCRIPTION
**Pull request by dlapidos-st**
- Add metadata columns to users and outbox_events tables
- Cast `unit` as `int` when passing as Postgres parameters

### CODE REVIEW

File: scripts/sql/0002-add-metadata-cols-to-users.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE public.users 
+    ADD COLUMN IF NOT EXISTS created_at timestamp DEFAULT (now() at time zone 'utc') NOT NULL;
```
Comment (atamelo-st) [inline scripts/sql/0002-add-metadata-cols-to-users.sql:2]: Nit: Letting the DB server deciding on the default doesn't sit well with me :) Plus, I do not see the real value from doing that. Imo it should solely be encapsulated within the DAL.

File: scripts/sql/0003-add-metadata-cols-to-outbox-events.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE public.outbox_events 
+    ADD COLUMN IF NOT EXISTS timestamp timestamp DEFAULT (now() at time zone 'utc') NOT NULL;
```
Comment (atamelo-st) [inline scripts/sql/0003-add-metadata-cols-to-outbox-events.sql:2]: The same with default.

File: scripts/sql/0003-add-metadata-cols-to-outbox-events.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+ALTER TABLE public.outbox_events 
+    ADD COLUMN IF NOT EXISTS aggregate_version integer DEFAULT 0 NOT NULL;
```
Comment (atamelo-st) [inline scripts/sql/0003-add-metadata-cols-to-outbox-events.sql:5]: The same concern with default.

File: scripts/sql/0003-add-metadata-cols-to-outbox-events.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+ALTER TABLE public.outbox_events 
+    ADD COLUMN IF NOT EXISTS event_schema_version integer DEFAULT 0 NOT NULL;
```
Comment (atamelo-st) [inline scripts/sql/0003-add-metadata-cols-to-outbox-events.sql:8]: The same concern with default.

File: scripts/sql/0002-add-metadata-cols-to-users.sql (hunk 1 above)
Comment (dlapidos-st) [reply scripts/sql/0002-add-metadata-cols-to-users.sql:2]: yeah, I just did this to maintain backwards compatibility with the API, as Postgres will populate the column with the `DEFAULT`

**Pull request merged**